### PR TITLE
x-pack/filebeat/input/http_endpoint: Add validation for empty URL config

### LINKED ISSUE #36772: x-pack/filebeat/input/http_endpoint: config.URL can be empty resulting in panic
We have seen an integration using the http_endpoint input that fails with a panic due to a misconfiguration.
```
Input 'http_endpoint' failed with: input panic with: http: invalid pattern
goroutine 197 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/elastic/beats/v7/filebeat/input/v2/input-stateless.configuredInput.Run.func1()
	github.com/elastic/beats/v7/filebeat/input/v2/input-stateless/stateless.go:81 +0x7f
panic({0x55fbecb7bfc0, 0x55fbed39ef80})
	runtime/panic.go:884 +0x212
net/http.(*ServeMux).Handle(0xc000bb8780, {0x0, 0x0}, {0x55fbed3a4680?, 0xc00076de60})
	net/http/server.go:2497 +0x25f
github.com/elastic/beats/v7/x-pack/filebeat/input/http_endpoint.(*pool).serve(_, {0xc000bba830, {0xc000253780, 0x3f}, {{0x55fbebe57b10, 0x8}, {0x55fbebe57b10, 0x8}, {0x55fbebe4fba1, 0x5}, ...}, ...}, ...)
	github.com/elastic/beats/v7/x-pack/filebeat/input/http_endpoint/input.go:140 +0x63f
github.com/elastic/beats/v7/x-pack/filebeat/input/http_endpoint.(*httpEndpoint).Run(_, {0xc000bba830, {0xc000253780, 0x3f}, {{0x55fbebe57b10, 0x8}, {0x55fbebe57b10, 0x8}, {0x55fbebe4fba1, 0x5}, ...}, ...}, ...)
	github.com/elastic/beats/v7/x-pack/filebeat/input/http_endpoint/input.go:88 +0x6c
github.com/elastic/beats/v7/filebeat/input/v2/input-stateless.configuredInput.Run({{_, _}}, {0xc000bba830, {0xc000253780, 0x3f}, {{0x55fbebe57b10, 0x8}, {0x55fbebe57b10, 0x8}, {0x55fbebe4fba1, ...}, ...}, ...}, ...)
	github.com/elastic/beats/v7/filebeat/input/v2/input-stateless/stateless.go:97 +0x24b
github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start.func1()
	github.com/elastic/beats/v7/filebeat/input/v2/compat/compat.go:114 +0x215
created by github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start
	github.com/elastic/beats/v7/filebeat/input/v2/compat/compat.go:111 +0xd8
```
This happens when the `url` config option is empty. The default value set for this is "/", however, in this case that appears not to have been set. The validation code should check that the config is correct in this field to provide a more user-friendly error in the logs.

### PR DESCRIPTION
**Pull request by bhapas**
## Proposed commit message

Integrations using the `http_endpoint` input that fails with a panic due to a misconfiguration. Validation added and a user-friendly log is added.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #36772

### CODE REVIEW

File: x-pack/filebeat/input/http_endpoint/config.go [hunk 1]
```diff
@@ -110,6 +110,10 @@ func (c *config) Validate() error {
 		return errors.New("crc.provider is required when crc.secret is defined")
 	}
 
+	if c.URL == "" {
```
Comment (andrewkroh) [inline x-pack/filebeat/input/http_endpoint/config.go:113]: The `URL` field can be marked with a struct tag.

```
URL                   string                  `config:"url" validate:"required"`
```

See https://pkg.go.dev/github.com/elastic/go-ucfg#Config.Unpack

Comment (bhapas) [reply x-pack/filebeat/input/http_endpoint/config.go:113]: And continue to d ovalidation for `c.URL == ""` to return custom error message?

File: x-pack/filebeat/input/http_endpoint/config_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+				ResponseBody: "",
+				Method:       http.MethodPost,
+			},
+			wantError: fmt.Errorf("response_body must be valid JSON"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.config.URL = ""
```
Comment (andrewkroh) [inline x-pack/filebeat/input/http_endpoint/config_test.go:52]: This won't test the behavior of the `validate` struct tags. That executes when config.Unpack() is called. So if the goal of this test was to unit test your Validate function then its good. 

But if you want test that all of the validation associated with this struct works when Unpack is called, then I would change the test to use `Unpack`.

```diff
diff --git a/x-pack/filebeat/input/http_endpoint/config.go b/x-pack/filebeat/input/http_endpoint/config.go
index 92385a33b1..e073302ef1 100644
--- a/x-pack/filebeat/input/http_endpoint/config.go
+++ b/x-pack/filebeat/input/http_endpoint/config.go
@@ -51,8 +51,6 @@ func defaultConfig() config {
 	return config{
 		Method:        http.MethodPost,
 		BasicAuth:     false,
-		Username:      "",
-		Password:      "",
 		ResponseCode:  200,
 		ResponseBody:  `{"message": "success"}`,
 		ListenAddress: "127.0.0.1",
@@ -60,14 +58,6 @@ func defaultConfig() config {
 		URL:           "/",
 		Prefix:        "json",
 		ContentType:   "application/json",
-		SecretHeader:  "",
-		SecretValue:   "",
-		HMACHeader:    "",
-		HMACKey:       "",
-		HMACType:      "",
-		HMACPrefix:    "",
-		CRCProvider:   "",
-		CRCSecret:     "",
 	}
 }
 
@@ -110,10 +100,6 @@ func (c *config) Validate() error {
 		return errors.New("crc.provider is required when crc.secret is defined")
 	}
 
-	if c.URL == "" {
-		return fmt.Errorf("webhook path URL can not be empty")
-	}
-
 	return nil
 }
 
diff --git a/x-pack/filebeat/input/http_endpoint/config_test.go b/x-pack/filebeat/input/http_endpoint/config_test.go
index f5318e214b..8c285422ec 100644
--- a/x-pack/filebeat/input/http_endpoint/config_test.go
+++ b/x-pack/filebeat/input/http_endpoint/config_test.go
@@ -5,27 +5,28 @@
 package http_endpoint
 
 import (
-	"fmt"
 	"net/http"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	confpkg "github.com/elastic/elastic-agent-libs/config"
 )
 
 func Test_validateConfig(t *testing.T) {
 	testCases := []struct {
 		name      string // Sub-test name.
 		config    config // Load config parameters.
-		wantError error  // Expected error
+		wantError string // Expected error
 	}{
 		{
-			name: "invalid URL",
+			name: "empty URL",
 			config: config{
 				URL:          "",
 				ResponseBody: `{"message": "success"}`,
 				Method:       http.MethodPost,
 			},
-			wantError: fmt.Errorf("webhook path URL can not be empty"),
+			wantError: "string value is not set accessing 'url'",
 		},
 		{
 			name: "invalid method",
@@ -34,27 +35,28 @@ func Test_validateConfig(t *testing.T) {
 				ResponseBody: `{"message": "success"}`,
 				Method:       "random",
 			},
-			wantError: fmt.Errorf("method must be POST, PUT or PATCH: random"),
+			wantError: "method must be POST, PUT or PATCH: random",
 		},
 		{
-			name: "invalid URL",
+			name: "invalid ResponseBody",
 			config: config{
 				URL:          "/",
 				ResponseBody: "",
 				Method:       http.MethodPost,
 			},
-			wantError: fmt.Errorf("response_body must be valid JSON"),
+			wantError: "response_body must be valid JSON",
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			tc.config.URL = ""
-			// Execute config validation
-			err := tc.config.Validate()
+			c := confpkg.MustNewConfigFrom(tc.config)
+
+			config := defaultConfig()
+			err := c.Unpack(&config)
 
 			// Validate responses
-			assert.Equal(t, tc.wantError, err)
+			assert.Contains(t, err.Error(), tc.wantError)
 		})
 	}
 }

```

File: x-pack/filebeat/input/http_endpoint/config.go (hunk 1 above)
Comment (andrewkroh) [reply x-pack/filebeat/input/http_endpoint/config.go:113]: It won't be necessary.

File: x-pack/filebeat/input/http_endpoint/config_test.go (hunk 2 above)
Comment (bhapas) [reply x-pack/filebeat/input/http_endpoint/config_test.go:52]: Got it..! Thanks Andrew

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**